tests/test_deploy_github_pages.py

```diff
@@ -0,0 +1,220 @@
+    current_commit_id = run(["git", "ls-remote", f"https://{user_access_token}@github.com/exasol/sphinx-github-pages-generator-test.git", target_branch], capture_output=True, check=True)
+    commit_id_old = current_commit_id.stdout
+
+    deploy_github_pages.deploy_github_pages(["--target_branch", target_branch,
+                                             "--push_origin", "origin",
+                                             "--push_enabled", "push",
+                                             "--source_branch", source_branch,
+                                             "--source_dir", cwd,
+                                             "--module_path", ["../test_package", "../another_test_package"]])
+    current_commit_id = run(["git", "ls-remote", f"https://{user_access_token}@github.com/exasol/sphinx-github-pages-generator-test.git", target_branch], capture_output=True, check=True)
+    commit_id_new = current_commit_id.stdout
+
+    # the docu files where not updated, so no new commit should be pushed to the remote
+    assert commit_id_old != 0
+    assert commit_id_old == commit_id_new
+    assert not commit_id_new == ""
+    remove_branch(target_branch)
+
+
+def test_verify_existence_of_generated_files_on_remote_after_push(setup_test_env):
+    _, _, cwd = setup_test_env
+    # generate files locally in test, and with generator, look at diff between local files and remote files?
+
+    source_branch = "5-add-tests"
+    run(["git", "checkout", source_branch], check=True)
+    target_branch = "test-docu-new-branch"
+    remove_branch(target_branch)
+    module_path = ["../test_package", "../another_test_package"]
+    deploy_github_pages.deploy_github_pages(["--target_branch", target_branch,
+                                             "--push_origin", "origin",
+                                             "--push_enabled", "push",
+                                             "--source_branch", source_branch,
+                                             "--source_dir", cwd,
+                                             "--module_path", module_path])
+
+    with TemporaryDirectory() as tempdir2:
+        os.chdir(tempdir2)
+        user_name, user_access_token, cwd = setup_workdir()
+        source_branch = "5-add-tests"
+        run(["git", "checkout", source_branch], check=True)
+        build_dir = tempdir2 + "/build"
+        intermediate_dir = tempdir2 + "/intermediate"
+        for module in module_path:
+            run(["sphinx-apidoc", "-T", "-e", "-o", "api", module], check=True)
+        run(["sphinx-build", "-b", "html", "-d", intermediate_dir, "-W", cwd, build_dir], check=True)
+        from pathlib import Path
+        from shutil import rmtree
+        output_dir = Path("./")
+        print(f"Copying HTML output {build_dir} to the output directory {output_dir}")
+        for obj in os.listdir(build_dir):
+            if ".doctree" in str(obj):
+                try:
```

Comment:
Here you still use .doctree.

Actually a test would be good which checks that .doctree is not part of the output directory.

Reply:
I thought i fixed that already ...

fixed, and additional test added.